angular-redactor-2.js

```diff
@@ -0,0 +1,62 @@
+(function() {
+    'use strict';
+
+    /**
+     * usage: <textarea ng-model="content" redactor></textarea>
+     *
+     *    additional options:
+     *      redactor: hash (pass in a redactor options hash)
+     *
+     */
+
+    var redactorOptions = {};
+
+    angular.module('angular-redactor', [])
+        .constant('redactorOptions', redactorOptions)
+        .directive('redactor', ['$timeout', function($timeout) {
+            return {
+                restrict: 'A',
+                require: 'ngModel',
+                link: function(scope, element, attrs, ngModel) {
+
+                    // Expose scope var with loaded state of Redactor
+                    scope.redactorLoaded = false;
+
+                    var updateModel = function updateModel(value) {
+                            // $timeout to avoid $digest collision
+                            $timeout(function() {
+                                scope.$apply(function() {
+                                    ngModel.$setViewValue(value);
+                                });
+                            });
+                        },
+                        options = {
+                            callbacks: {
+                                change: updateModel
+                            }
+                        },
+                        additionalOptions = attrs.redactor ?
+                            scope.$eval(attrs.redactor) : {},
+                        editor;
+
+                    angular.extend(options, redactorOptions, additionalOptions);
+
```

Comment:
How come you removed the code that keeps the custom callback function that you can pass in on initialization? It would also have to be updated with redactor's new callback structure, but here's the updated code.

```javascript
// prevent collision with the constant values on callbacks.change
var changeCallback = additionalOptions.callbacks.change || redactorOptions.callbacks.change;
if (changeCallback) {
    options = {
        callbacks: {
          change: function(value) {
              updateModel.call(this, value);
              changeCallback.call(this, value);
          }
        }
    }
}
```